### DISCUSSION TITLE
DSCMD server for local PowerBI

### DISCUSSION THREAD

**Discussion by vhunyad** [DaxStudio/DaxStudio#1317] [Q&A] [answered]
What is the correct --server option for a local PowerBI Desktop? I can not figure it out.

**Comment by dgosbell** [comment, marked as answer]
At the moment you would have to use localhost:<port> the problem is that Desktop uses a random <port> each time it starts up. The connection dialog in the DAX Studio UI can find the port for you - what I want to do for dscmd is to extend it so that you can just pass in the name of the PBIX file and let dscmd find the port for you, but it does not do this currently

**Reply by vhunyad** [reply]
Thanks. Finally I solved it with a python script which checks msmdsrv.exe' open connections, call each port with a simple DB specific query, store the result in a file and evaluate the file to identify which database is on which port. I regularly have multiple PBX's open, so just having a single port is not enough.
Calling just the pbx name would be much simpler.

**Reply by iveniveniveniven** [reply]
@vhunyad Could you please share your python script? I have a similar issue. Thanks and all the best!

**Reply by dgosbell** [reply]
As of the 3.3.0 release dscmd supports connecting to an open pbix file with a format like --server <filename>.pbix as documented here https://daxstudio.org/docs/features/command-line/connecting/#power-bi-desktop note that <filename> should just be the name of the file (surrounded by quotes if it includes spaces or special characters) do not include the full path.

**Comment by dataflipco** [comment]
To connect Power BI Desktop to a local data source, you don't need to use a server option like in command-line tools. Power BI Desktop simplifies this process with its built-in interface. To connect to a local SQL Server or other data sources, just open Power BI Desktop, click on Home > Get Data > SQL Server (or choose the relevant data source). In the Server field, enter your local server's name, which might be localhost or .\SQLEXPRESS if you're using SQL Server Express. After that, you can provide the database name, credentials if necessary, and click OK. Power BI will then establish the connection without needing any extra command-line options.

**Reply by dgosbell** [reply]
Thanks for you suggestion. But this question is not about connecting out from Power BI Desktop it is about connecting in using the dscmd.exe tool that ships with DAX Studio.
